### PR TITLE
Closes #3467. Add benchmarks for Sapling spends and outputs.

### LINKED ISSUE #3467: Add Sapling benchmarks
The [benchmark suite](https://speed.z.cash/timeline/) has time and memory benchmarks for creating and verifying Sprout JoinSplits, but not for creating and verifying Sapling Spends or Outputs.

### PR DESCRIPTION
**Pull request by bitcartel**
Four new benchmarks are added to RPC zcbenchmark:
- createsaplingspend
- createsaplingoutput
- verifysaplingspend
- verifysaplingoutput

### CODE REVIEW

File: src/zcbenchmarks.cpp [hunk 1]
```diff
@@ -472,3 +474,159 @@ double benchmark_listunspent()
     auto unspent = listunspent(params, false);
     return timer_stop(tv_start);
 }
+
+extern double benchmark_create_sapling_spend()
```
Comment (Eirik0) [inline src/zcbenchmarks.cpp]: We seem to compile just fine w/o the `extern` keyword here (and below). Is there any reason to have this here?

Comment (bitcartel) [reply src/zcbenchmarks.cpp]: It's a copy and paste from the header file... good spot.. removing.

**Comment by bitcartel** [comment]
@zkbot r+

**Comment by bitcartel** [comment]
Transient failure.  Trying again.  @zkbot retry

**Comment by arielgabizon** [comment]
Should these also be added somehow to `performance-measurements.sh`?

**Pull request merged**